src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -26,24 +27,42 @@
     private long threadLocalSetupTimeDiff = 0;
     private long pairHMMSetupTime = 0;
 
+    public enum Implementation {AVX, OMP}
+
     private final PairHMMNativeBinding pairHmm;
 
     //Hold the mapping between haplotype and index in the list of Haplotypes passed to initialize
     //Use this mapping in computeLikelihoods to find the likelihood value corresponding to a given Haplotype
     private final Map<Haplotype, Integer> haplotypeToHaplotypeListIdxMap = new LinkedHashMap<>();
     private HaplotypeDataHolder[] mHaplotypeDataArray;
 
-    public VectorLoglessPairHMM(PairHMMNativeArguments args) throws UserException.HardwareFeatureException {
-        // TODO: connect GATK temp directory
-        final boolean isSupported = new IntelPairHmm().load(null);
-
-        if (!isSupported) {
-            throw new UserException.HardwareFeatureException("Machine does not support AVX PairHMM.");
+    public VectorLoglessPairHMM(Implementation implementation,
+                                PairHMMNativeArguments pairHmmArgs) throws UserException.HardwareFeatureException {
+        final boolean isSupported;
+
+        switch (implementation) {
+            case AVX:
+                isSupported = new IntelPairHmm().load();
+                if (!isSupported) {
+                    throw new UserException.HardwareFeatureException("Machine does not support AVX PairHMM.");
+                }
+                pairHmm = new IntelPairHmm();
+                break;
+
+            case OMP:
+                isSupported = new IntelPairHmmOMP().load();
+                if (!isSupported) {
+                    throw new UserException.HardwareFeatureException("Machine does not support OpenMP AVX PairHMM.");
+                }
+                pairHmm = new IntelPairHmmOMP();
+                logger.info("Maximum number of OpenMP threads for AVX : " + pairHmmArgs.maxNumberOfThreads);
```

Comment:
Actually, I'm curious why this didn't cause an NPR in the tests.  From the line coverage it looks like `isSupported` was true, but then `new IntelPairHmmOMP()` threw an exception and it never reached the line below.  Is it possible that it could claim to load and then throw a hardware feature exception on instantiation?